vars/buildPlugin.groovy

```diff
@@ -106,11 +107,18 @@ def call(Map params = [:]) {
                                     mavenOptions += "-DskipTests"
                                 }
                                 mavenOptions += "clean install"
+                                if (!skipTests && first && enableCoverage) {
+                                    mavenOptions += "jacoco:prepare-agent test jacoco:report"
```

Comment:
shouldn't this be done automatically as part of the maven build, using the recommended configuration in the code-coverage-api? 

(skip to jacoco, no anchor unfortunately)
https://github.com/jenkinsci/code-coverage-api-plugin#1-config-coverage-tool-to-generate-reports

Reply:
Yes, this would be another way to do it. But how do we toggle the execution of Jacoco then? Should we assume that we always record the coverage and remove all toggles?

Reply:
is there any reason not to?

Reply:
Normally, I run test twice: once without coverage in order to make sure that the unpatched class files are tested, and a second time with coverage. This duplicates the run time for the tests. But maybe this approach is too academic since in practice I never noticed tests that failed only due the coverage modifications off the class files.

Reply:
I come from a gradle world where everything like this is just bound to the `check` phase.

It seems odd to me to have to directly invoke plugins, I don't really understand your answer on why you run it without first.

If it adds quite a bit of time to the build I would understand it, although I tried it out on my plugins and it seemed fine.

Reply:
> I don't really understand your answer on why you run it without first.

Because the coverage step changes the class files. I.e. the tests may theoretically fail (or succeed) because of the changed class files. Normally we should test the same executable that we are shipping. (As noted, this might be a too academical approach.(

Reply:
Do we already have a helper method that detects if files of a given pattern exist?

Reply:
Not that I’m aware of